flocx_market/db/sqlalchemy/models.py

```diff
@@ -45,7 +46,16 @@ class Bid(Base):
     @orm.validates('cost')
     def validate_cost(self, key, value):
         if value < 0:
-            raise ValueError('Cost must be >= 0')
+            raise e.InvalidInput('Cost must be >= 0')
+        return value
+
+    @orm.validates('server_config_query')
+    def validate_server_config_query(self, key, value):
+        if type(value) is not dict:
+            raise e.InvalidInput('server_config_query must be a dictionary')
+        if 'specs' not in value:
+            raise e.InvalidInput("server_config_query must contain key \
```

Comment:
Minor nit: you don't need `\` for extending the line here. You can just write:

```
             raise e.InvalidInput("server_config_query must contain key "
                                  "'specs'")
```

(Because multiple adjacent strings are concatentated, as in `print('this ' 'is a ' 'test')`.)